mitodl/mitxpro
# Global coupons/promos #62

#### Pre-Flight checklist


#### What are the relevant tickets?
[Ticket](https://trello.com/c/CWR4dPwy/62-apply-coupons-to-all-products)

#### What's this PR do?
Adds an `is_global` attribute to `Coupon` model which signifies that a coupon is applicable to all products.
When the coupon is set to global:
- Product type and product picker is hidden
- Validation on at least one product being required is removed


#### How should this be manually tested?
Go to `/ecommerce/admin/coupons` to create a coupon with `is_global` set to true. Try to checkout any product with this coupon, it should be applicable on all. 

#### Screenshots (if appropriate)
<img width="878" alt="Screenshot 2020-03-03 at 8 15 34 PM" src="https://user-images.githubusercontent.com/6387678/75789633-ce3a8380-5d8b-11ea-8620-5e1f26bb4abc.png">
